### PR TITLE
Add automatic documentation generation

### PR DESCRIPTION
**Pull request by mrbean-bremen**
Related to #1199.

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [ ] I have updated API documentation
- [ ] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- first draft to add automatic documentation generation in GitHub Pages

### CODE REVIEW

File: Documentation/log.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{"message":"Invalid file link:(~/usage/logging_config.html).","source":"BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.TocDocumentProcessor.Save","file":"usage/toc.yml","line":"0","date_time":"2021-09-20T20:23:15.8938681Z","message_severity":"warning","code":"InvalidFileLink","correlation_id":"F1ABF3B6-A6EE-403F-AA56-0451A0F01872.2.179.1.32.9.1.19"}
+{"message":"Invalid file link:(~/usage/raw_pixel_data.html).","source":"BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.TocDocumentProcessor.Save","file":"usage/toc.yml","line":"0","date_time":"2021-09-20T20:23:15.8938681Z","message_severity":"warning","code":"InvalidFileLink","correlation_id":"F1ABF3B6-A6EE-403F-AA56-0451A0F01872.2.179.1.32.9.1.20"}
+{"message":"Invalid file link:(~/usage/prerequisites.html).","source":"BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.TocDocumentProcessor.Save","file":"usage/toc.yml","line":"0","date_time":"2021-09-20T20:23:15.8938681Z","message_severity":"warning","code":"InvalidFileLink","correlation_id":"F1ABF3B6-A6EE-403F-AA56-0451A0F01872.2.179.1.32.9.1.21"}
```
Comment (mrbean-bremen) [inline Documentation/log.txt:55]: Was this added intentionally?

File: Documentation/FO-DICOM.Documentation.DocFx.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="docfx.console" Version="2.58.4">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+
+</Project>
```
Comment (mrbean-bremen) [inline Documentation/FO-DICOM.Documentation.DocFx.csproj:16]: Thanks - I think this is a good idea! I had tried to add an empty project which does nothing, but actually building the documentation is better.

File: Documentation/log.txt (hunk 1 above)
Comment (gofal) [reply Documentation/log.txt:55]: oh, no. this is the log output of the docfx engine. I will remove it and add it to .gitignore

File: Documentation/FO-DICOM.Documentation.DocFx.csproj (hunk 2 above)
Comment (gofal) [reply Documentation/FO-DICOM.Documentation.DocFx.csproj:16]: when you build this project including this nugetpackage from docfx, then the docfx engine creates the files locally and the asp.net core webserver serves them, so you can browse it out of visual studio to check if everything works fine. Found this on the docfx documentation page.

Comment (mrbean-bremen) [reply Documentation/FO-DICOM.Documentation.DocFx.csproj:16]: I just tried it - this is really nice, I didn't see that in the documentation, thanks!

**Pull request merged**